src/os/ui/areas.js

```diff
@@ -167,7 +183,106 @@ os.ui.AreasCtrl.prototype.export = function() {
         return null;
       }).filter(os.fn.filterFalsey);
 
-  os.ui.ex.AreaExportCtrl.start(areas);
+  os.ui.ex.AreaExportCtrl.start(formattedAreas);
+};
+
+
+/**
+ * Events fired by the query area import menu.
+ * @enum {string}
+ */
+os.ui.AreasCtrl.EventType = {
+  ALL: 'areaexport:all',
+  SELECTED: 'areaexport:selected',
+  ACTIVE: 'areaexport:active'
+};
+
+
+/**
+ * Opens the area export menu.
+ *
+ * @export
+ */
+os.ui.AreasCtrl.prototype.openExportMenu = function() {
+  var target = this.element.find('.js-export-group');
+  var menu = new os.ui.menu.Menu(new os.ui.menu.MenuItem({
+    type: os.ui.menu.MenuItemType.ROOT,
+    children: [{
+      label: 'Export All',
+      eventType: os.ui.AreasCtrl.EventType.ALL,
+      tooltip: 'Export all areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
+      sort: 1
+    }, {
+      label: 'Export Selected',
+      eventType: os.ui.AreasCtrl.EventType.SELECTED,
+      tooltip: 'Export only selected areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
+      sort: 2
+    }, {
+      label: 'Export Active',
+      eventType: os.ui.AreasCtrl.EventType.ACTIVE,
+      tooltip: 'Export only active areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
```

Comment:
You can get rid of the menu context object entirely if you switch `os.ui.AreasCtrl.exportEventHandler_` to be on the prototype and use `this.exportEventHandlier_.bind(this)`. This is a cleaner way than passing around that context object and will allow you to directly reference `this.scope['areas/activeAreas/selected']` from the handler function.